pykechain/models/widgets/widgets_manager.py

```diff
@@ -356,8 +367,10 @@ def add_supergrid_widget(
             }
         )
 
-        if parent_instance:
-            meta[AssociatedObjectId.PARENT_INSTANCE_ID] = parent_instance
+        if parent_instance_id:
+            meta[AssociatedObjectId.PARENT_INSTANCE_ID] = parent_instance_id
+        if parent_model_id:
+            meta[AssociatedObjectId.PARENT_MODEL_ID] = parent_model_id
```

Comment:
Why the if? parent_model_id should be provided in all scenarios.

Reply:
yup